include/mbgl/gfx/backend.hpp

```diff
@@ -0,0 +1,44 @@
+#pragma once
+
+#include <cstdint>
+#include <cstdlib>
+#include <memory>
+
+namespace mbgl {
+namespace gfx {
+
+class Backend {
+public:
+    enum class Type : uint8_t {
+        OpenGL,
+    };
+
+    static constexpr Type DefaultType = Type::OpenGL;
+
+    static void SetType(const Type value) {
+        if (Value(value) != value) {
+            abort(); // SetType must be called prior to any GetType calls.
+        }
```

Comment:
```
Value(value);
assert(Value(value) == value);
```
?

Reply:
there's no way out if this assert is false, so we can just as well run `abort`